api/src/pcapi/scripts/add_venue_providers_on_venue_with_siret/main.py

```diff
@@ -0,0 +1,344 @@
+    subquery = (
+        db.session.query(Venue.managingOffererId)
+        .filter(Venue.siret.is_not(None))
+        .group_by(Venue.managingOffererId)
+        .having(sa.func.count(Venue.id) == 1)
+        .subquery()
+    )
+    return (
+        db.session.query(Venue)
+        .join(providers_models.VenueProvider)
+        .join(subquery, Venue.managingOffererId == subquery.c.managingOffererId)
+        .filter(Venue.siret.is_(None), Venue.isPermanent.is_(True))
+        .options(sa_orm.load_only(Venue.id, Venue.siret, Venue.managingOffererId))
+        .all()
+    )
+
+
+def _get_unique_venues_with_siret_by_offerer() -> list[Venue]:
+    VenueAlias = sa_orm.aliased(Venue)
+    subquery = (
+        db.session.query(Venue.managingOffererId)
+        .filter(Venue.siret.is_not(None))
+        .group_by(Venue.managingOffererId)
+        .having(sa.func.count(Venue.id) == 1)
+        .subquery()
+    )
+    return (
+        db.session.query(Venue)
+        .join(subquery, Venue.managingOffererId == subquery.c.managingOffererId)
+        .filter(Venue.siret.is_not(None))
+        .filter(
+            sa.not_(
+                db.session.query(VenueAlias)
+                .filter(
+                    VenueAlias.managingOffererId == Venue.managingOffererId,
+                    VenueAlias.id != Venue.id,
+                    VenueAlias.siret.is_not(None),
+                )
+                .exists()
+            )
+        )
+        .all()
+    )
+
+
+@atomic()
+def main(not_dry: bool) -> None:
+    log_modifications: list[tuple[int, str]] = []
+    log_fails: list[tuple[int, str]] = []
+    venues_without_siret_with_providers = _get_venues_without_siret_and_not_permanent_with_providers()
+    venues_with_siret_by_offerer = _get_unique_venues_with_siret_by_offerer()
+    user = User.query.get(MIGRATION_AUTHOR_ID)
+    venue_with_siret = {venue.managingOffererId: venue for venue in venues_with_siret_by_offerer}
+    for venue_source in venues_without_siret_with_providers:
+        venue_destination = venue_with_siret.get(venue_source.managingOffererId)
+        if not venue_destination:
+            log_fails.append((venue_source.id, f"Pas de siret sur cet offerer {venue_source.managingOffererId}"))
+            continue
+
+        for venue_provider in venue_source.venueProviders:
+            try:
+                with atomic():
+                    _create_venue_provider(venue_source, venue_destination, venue_provider, user)
+                    log_modifications.append(
+                        (
+                            venue_source,
+                            f"Ajout du provider {venue_provider.provider} sur la venue {venue_destination}",
+                        )
+                    )
+                    logger.info(
+                        "Adding provider %s on venue %s",
+                        venue_provider.providerId,
+                        venue_destination.id,
+                    )
```

Comment:
Vu qu'il y a déjà un décorateur `@atomic()` sur `_create_venue_provider`, pas besoin du `with atomic()` non ?
```suggestion
                _create_venue_provider(venue_source, venue_destination, venue_provider, user)
                log_modifications.append(
                    (
                        venue_source,
                        f"Ajout du provider {venue_provider.provider} sur la venue {venue_destination}",
                    )
                )
                logger.info(
                    "Adding provider %s on venue %s",
                    venue_provider.providerId,
                    venue_destination.id,
                )
```